meson.build

```diff
@@ -120,8 +120,12 @@ if libunibreak_dep.found()
     conf.set('CONFIG_UNIBREAK', 1)
 endif
 
-if get_option('test')
-    test_deps += dependency('libpng', version: '>= 1.2.0')
+png_dep = dependency(
+    'libpng',
+    version: '>= 1.2.0',
+    required: get_option('test').enabled() or get_option('compare').enabled())
+if png_dep.found()
+    png_deps += png_dep
```

Comment:
This is suboptimal, it will also try to look up libpng and show messages about it to the builder even when no enabled build target uses it. Instead the lookup should be dependend on users being enabled like before, and the dep then being required *(the default iirc)*.

Reply:
Does this really matter? It's a one-time cached configure check (and tbh we might end up turning the test/compare/etc programs on by default behind a libpng-availability gate at some point anyway).